File: apps/ios/Twende/Twende/Repositories/RatingRepository.swift
import Foundation

final class RatingRepository {
    static let shared = RatingRepository()
    private let api = APIService.shared

    private init() {}

    func submitRating(journeyId: String, driverId: String, score: Int, comment: String?) async throws {
        let body = RatingRequest(journeyId: journeyId, driverId: driverId, score: score, comment: comment)
        let _: APIResponse<Rating> = try await api.post("ratings", body: body)
    }

    func getDriverRating(driverId: String) async throws -> Double {
        let response: APIResponse<Rating> = try await api.get("drivers/\(driverId)/rating")
        return Double(response.data?.score ?? response.data?.stars ?? 0)
    }
}

File: apps/android/app/src/main/java/com/twende/app/data/repository/RatingRepository.kt
package com.twende.app.data.repository

import com.twende.app.data.api.TwendeApi
import com.twende.app.data.model.Rating
import com.twende.app.data.model.RatingRequest
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class RatingRepository @Inject constructor(
    private val api: TwendeApi,
) {
    suspend fun submitRating(
        journeyId: String,
        driverId: String,
        score: Int,
        comment: String? = null,
    ): Result<Rating> = runCatching {
        val response = api.submitRating(
            RatingRequest(
                journeyId = journeyId,
                driverId = driverId,
                score = score,
                comment = comment,
            )
        )
        if (response.isSuccessful) {
            val body = response.body()!!
            if (body.success && body.data != null) {
                body.data
            } else {
                throw Exception(body.error?.message ?: "Rating submission failed")
            }
        } else {
            throw Exception("Rating submission failed: ${response.code()}")
        }
    }

    suspend fun getDriverRating(driverId: String): Result<Double> = runCatching {
        val response = api.getDriverRating(driverId)
        if (response.isSuccessful) {
            val body = response.body()!!
            if (body.success && body.data != null) {
                body.data
            } else {
                throw Exception(body.error?.message ?: "Failed to load driver rating")
            }
        } else {
            throw Exception("Failed to load driver rating: ${response.code()}")
        }
    }
}
